2025-06-02-default-reek-configuration.md

```diff
@@ -0,0 +1,105 @@
+# Default Reek Configuration
+
+## The Problem
+
+We like to use Reek in many of our projects as one of the tools to aid in improving the code quality, but we do not have a clear expected way to use Reek in the different projects we use and we have no clear guidelines about when to make exception or changes to its configuration.
+
+## Proposed initial Reek configuration
+
+This RFC is a proposal of a default Reek configuration file that we would use in established projects to reduce the number of warnings and the need of exceptions that we see with the [default configuration](https://github.com/troessner/reek/blob/a8649370ced77dd0e798b00dc97ff916f68bb002/docs/defaults.reek.yml), and to also define some guideline or expectation on how we can decide when to add exceptions and modifications for different needs.
+
+### Default .reek configuration file
+
+This file is meant to be added in the root of the project as `.reek.yml` to handle common cases of Reek warnings that will want either disabled or reconfigured based on what we've seen in many projects.
+
+It's a combination of the [official `rails-friendly` configuration recommended in the Reek's README](https://github.com/troessner/reek?tab=readme-ov-file#working-with-rails) and some changes to the default rules.
+
+```yml
+detectors:
```

Comment:
I think this is useful too.
It allows us to do `rescue e =>` and `map { |x| x.do_something }`

```
  UncommunicativeVariableName:
    accept:
     - e
     - x
```

Reply:
I wonder if we should add `k` and `v` as they are common short names for `key` and `value` when iterating over a hash

Reply:
Sounds good to me